### PR TITLE
Vde

### PR DESCRIPTION
**Pull request by ifurther**
#7118

### CODE REVIEW

File: packages/vde2/build.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+TERMUX_PKG_GIT_BRANCH=master
+TERMUX_PKG_SRCURL=https://github.com/virtualsquare/vde-2.git
+TERMUX_PKG_DEPENDS="libpcap,libtool , openssl, libc++"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" --disable-python"
+
+termux_step_pre_configure() {
+	autoreconf --install
+}
+CFLAGS+=" -Drindex=strrchr"
```
Comment (xeffyr) [inline packages/vde2/build.sh:16]: Just pasted it onto wrong line or why this is set between functions? Any environment configuration must be done in pre configure step.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/vde2/build.sh (hunk 1 above)
Comment (ifurther) [reply packages/vde2/build.sh]: OK, I moved it to pre configure

File: packages/vde2/build.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/virtualsquare/vde-2
+TERMUX_PKG_DESCRIPTION="Virtual Distributed Ethernet for emulators like qemu"
+TERMUX_PKG_LICENSE="custom"
+TERMUX_PKG_LICENSE_FILE="COPYING"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=2.3.2
+TERMUX_PKG_GIT_BRANCH=master
```
Comment (xeffyr) [inline packages/vde2/build.sh:7]: Never specify `master` or any other branch which does not describe package version.

Specify git tag there.

File: packages/vde2/build.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+TERMUX_PKG_SRCURL=https://github.com/virtualsquare/vde-2.git
+TERMUX_PKG_DEPENDS="libpcap,libtool , openssl, libc++"
+TERMUX_PKG_BUILD_IN_SRC=true
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" --disable-python"
+
+termux_step_pre_configure() {
+	autoreconf --install
+	CFLAGS+=" -Drindex=strrchr -DSTDC_HEADERS"
```
Comment (xeffyr) [inline packages/vde2/build.sh:15]: Needs fix for `index()` as well.

File: packages/vde2/build.sh [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+}
+termux_step_make(){
+	make V=1
```
Comment (xeffyr) [inline packages/vde2/build.sh:18]: Why it is necessary to override `termux_step_make()`? You can pass make flags through `TERMUX_PKG_EXTRA_MAKE_ARGS`.

**Review by xeffyr — CHANGES_REQUESTED** [review]

File: packages/vde2/build.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/virtualsquare/vde-2
+TERMUX_PKG_DESCRIPTION="Virtual Distributed Ethernet for emulators like qemu"
+TERMUX_PKG_LICENSE="custom"
```
Comment (Grimler91) [inline packages/vde2/build.sh:3]: Package seem to have three different licenses. Use custom, gpl-2.0, lgpl-2.1 (that's what [archlinux does](https://archlinux.org/packages/extra/x86_64/vde2/)) and set LICENSE_FILE to all three COPYING files

**Review by Grimler91 — CHANGES_REQUESTED** [review]

File: packages/vde2/build.sh (hunk 5 above)
Comment (xeffyr) [reply packages/vde2/build.sh:3]: If package uses custom license, even besides GPL, LGPL or other well known, there should be specified "custom".

Comment (Grimler91) [reply packages/vde2/build.sh:3]: Why? To me it looks like different part of the package have different licenses, most of it is packaged under GPL, but they have chose to license the shared libraries under LGPL, and the program slirpvde under a custom (modified BSD 3-clause) license. However, looks like slirpvde is linked against another GPL library in their source, so I suppose the GPL should apply to slirpvde as well (but I am not a lawyer or expert in licensing).

Specifying all three licenses (GPL, LGPL, custom), and packaging all three Copying files, gives the user the best idea about the licensing-status of the package IMO

Comment (Grimler91) [reply packages/vde2/build.sh:3]: Ok, only custom needed to pass lint script

**Review by xeffyr — APPROVED** [review]

**Pull request merged**